src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -0,0 +1,367 @@
+            return (T[]) copyArray(this.data, this.index);
+        }
+
+        for (int i = 0; i < this.index; i++) {
+            array[i] = (T) this.data[i];
+        }
+
+        array[this.index] = null;
+        return array;
+    }
+
+    /**
+     * @return an iterator over the elements in this list in proper sequence.
+     */
+    @Override
+    public Iterator<T> iterator() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns a list iterator over the elements in this list (in proper
+     * sequence).
+     */
+    @Override
+    public ListIterator<T> listIterator() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns a list iterator over the elements in this list (in proper
+     * sequence), starting at the specified position in the list.
+     */
+    @Override
+    public ListIterator<T> listIterator(final int index) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Removes the first occurrence of the specified element from this list, if
+     * it is present (optional operation).
+     */
+    @Override
+    public boolean remove(final Object element) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Removes the element at the specified index in this list.
+     *
+     * @param index the index.
+     */
+    @Override
+    public T remove(final int index) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Removes from this list all of its elements that are contained in the
+     * specified collection (optional operation).
+     */
+    @Override
+    public boolean removeAll(final Collection<?> collection) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Retains only the elements in this list that are contained in the
+     * specified collection (optional operation).
+     */
+    @Override
+    public boolean retainAll(final Collection<?> collection) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Checks if this list contains all of the elements of the collection.
+     *
+     * @param collection the collection.
+     * @return true if this list contains all of the elements of the collection.
+     */
+    @Override
+    public boolean containsAll(final Collection<?> collection) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * increase the size of data[].
+     */
+    private void increaseSize() {
+        int newSize = this.data.length * 2;
+        this.data = Arrays.copyOf(this.data, newSize);
+    }
+
+    private void checkIndex(final int i) {
+        if (i > this.index - 1 || i < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    private T[] copyArray(final T[] data, final int index) {
+        final T[] array = (T[]) new Object[index];
+
+        for (int i = 0; i < index; i++) {
+            array[i] = data[i];
```

Comment:
This.data[i]

Reply:
no here I mean data that is send as a param